hamp_radar/iquick.py

```diff
@@ -234,6 +233,157 @@ def postprocess_iq(ds):
     return ds.pipe(decode_time)
 
 
+def untangle_iqf(
+    files,
+    zarr_path,
+    pulse_dt=208e-9,
+    wait1_dt=1.0e-6,
+):
+    """
+    Takes a list of pds files and returns zarr files for frame and pulse data respectivel.
+
+    files: path to pds data
+    zarr_path: directory in which zarr files are to be written
+    (pulse_dt, wait1_dt): radar parameters for determining range
+    """
+    import xarray as xr
+    import numpy as np
+    import pandas as pd
+
+    c_air = 299792458.0 / 1.0003  # speed of light in air
+
+    first = read_iq(files[0])
+
+    tbeg = first.time[0]
+    fbeg = first.frm[0]
+    nfft = first.sizes["fft"]
+    ngate = first.sizes["range"]
+
+    last = read_iq(files[-1])
+    tau_frame = (last.time[-1] - tbeg) / (last.frm[-1] - fbeg + 1)
+
+    tau_pulse = tau_frame / nfft
+    dtfft = tau_pulse * xr.DataArray(np.arange(nfft), dims=("fft",))
+
+    for file in files:
+        radar = read_iq(file)
+        if radar.sizes["fft"] != nfft:
+            raise ValueError("fft dimension cannot change among files")
+        if radar.sizes["range"] != ngate:
+            raise ValueError("range dimension cannot change among files")
+        time_frame = tbeg + tau_frame * (radar.frm - fbeg)
+        print(f"First frame {radar.frm[0].values} of file {file}")
+        if np.all(np.diff(radar.time) > 0):
```

Comment:
This if clause could be written in guard clause style, just as above. This would increase readability and reduce the indentation depth by one.
```suggestion
        if not np.all(np.diff(radar.time) > 0):
            raise ValueError("frames do not increment monotonically")
```
Due to the `raise`, the function would be aborted anyways and the remaining code could stay unindented. This would keep the "normal" program flow on one indentation level.